lib/MusicBrainz/Server/Controller/Root.pm

```diff
@@ -199,7 +199,7 @@ sub begin : Private
     $c->stats->enable(1) if DBDefs->DEVELOPMENT_SERVER;
 
     # Can we automatically login?
-    if (!$c->user_exists) {
+    if (!$c->user) {
```

Comment:
The `user_exists` method does not actually try to restore the user; if there is a valid session, it will return true, even if the owner of the session has been deleted in the meantime. Checking early for this condition results in a clearer error message than when the issue is detected later inside the Redis access try/catch block.